sys/net/gnrc/application_layer/tftp/gnrc_tftp.c

```diff
@@ -0,0 +1,749 @@
+/*
+ * Copyright (C) 2015 Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @ingroup net_gnrc_tftp
+ * @{
+ *
+ * @file
+ *
+ * @author      Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <time.h>
+
+#include "net/gnrc/tftp.h"
+#include "net/gnrc/netapi.h"
+#include "net/gnrc/netreg.h"
+#include "net/gnrc/udp.h"
+#include "net/gnrc/ipv6.h"
+#include "random.h"
+
+#define ENABLE_DEBUG                (1)
+#include "debug.h"
+
+#if ENABLE_DEBUG
+/* For PRIu16 etc. */
+#include <inttypes.h>
+#endif
+
+#define ENABLE_TFTP_ERROR_SEND                 0
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define CT_HTONS(x)         ((          \
```

Comment:
Because there are no "compile time" HTONS NTONS macro's. Which means that I couldn't define my enums like:

	typedef enum {
		TO_RRQ          = CT_HTONS(1),      /**< Read Request */
		TO_RWQ          = CT_HTONS(2),      /**< Write Request */
		TO_DATA         = CT_HTONS(3),      /**< Data */
		TO_ACK          = CT_HTONS(4),      /**< Acknowledgment */
		TO_ERROR        = CT_HTONS(5),      /**< Error */
		TO_OACK         = CT_HTONS(6),      /**< Option ACK */
	} tftp_opcodes_t;

And don't have to swap the error and op codes runtime time. I could change them to network_uint16_t's and use defines, but this is easier for debugging :)